stress_test/controller.py

```diff
@@ -0,0 +1,338 @@
+            self.wait_until_up(420000)
+        else:
+            logging.info('[start_controller] Controller already started.')
+
+        self.status = 'STARTED'
+
+    def stop(self):
+        """Wrapper to the controller stop handler
+        """
+
+        self.status = 'STOPPING'
+        if check_status() == '1':
+            logging.info('[stop_controller] Stopping controller.')
+            common.command_exec_wrapper(
+                [self.stop_hnd],
+                '[controller_stop_handler]', self.ssh_pass)
+            util.process.wait_until_process_finishes(self.pid, self.ssh_pass)
+            self.status = 'STOPPED'
+
+            else:
+            logging.info('[stop_controller] Controller already stopped.')
+
+    def build(self):
+        """ Wrapper to the controller build handler
+        """
+        logging.info('[Controller] Building')
+        self.status = 'BUILDING'
+
+        common.command_exec_wrapper([self.build_hnd],
+                             '[controller_build_handler]', self.ssh_pass)
+
+        self.status = 'BUILT'
+
+    def wait_until_listens(self,timeout_ms):
+        """ Waits for controller to start listening on specified port.
+
+        :param timeout_ms: milliseconds to wait (in milliseconds).
+        :raises Exception: If controller fails to start or if another process
+        listens on controllers port.
+        :rtype int
+        :type timeout_ms: int
+        """
+
+        logging.info('[Controller] Waiting to start listening on a port')
+
+        timeout = time.time() + (float(timeout_ms) / 1000)
+        while time.time() < timeout:
+            time.sleep(1)
+            gpid = util.process.getpid_listeningonport(self.port, self.ssh_pass)
+            logging.info('Returned pid listening on port {0}: {1}'.
+                          format(self.port, gpid))
+            if gpid == 0:
+                raise Exception('Another controller seems to have started in the '
+                                'meantime. Exiting...')
+
+        raise Exception('Controller failed to start within a period of {0} '
+                        'minutes'.format(timeout))
+
+
+    def wait_until_up(self,timeout_ms):
+        """ Waits for controller status to become 1 (started).
+
+        :param timeout_ms: milliseconds to wait (in milliseconds).
+        :raises Exception: If controller fails to start or if another process
+        listens on controllers port.
+        :type timeout_ms: int
+        """
+
+        logging.info('[Controller] Waiting to be started')
+
+        timeout = time.time() + (float(timeout_ms) / 1000)
+        while time.time() < timeout:
+            time.sleep(1)
+            if check_status() == '1':
+                return
+
+        raise Exception('Controller failed to start. '
+                        'Status check returned 0 after trying for {0} seconds.'.
+                        format(float(timeout_ms) / 1000))
+
+
+    @staticmethod
+    def new_controller(self)
```

Comment:
Since we will be calling a static method as `Classname.method()`, it would be wiser to rename this function to `new` so that we can call it like `Controller.new()` (and not `Controller.new_controller()`)